README.md
dunitgroup-dunit
================


ChangeLog
2003-05-08 01:00  juanco

	* want.xml, src/versioninfo.inc, src/versioninfo.rc,
	src/versioninfo.res, Contrib/XPGen/want.xml, src/versioninfo.inc:
	ready for new release

2003-03-20 13:57  juanco

	* src/: GUITestRunner.dfm, GUITestRunner.pas, QGUITestRunner.pas,
	QGUITestRunner.xfm, TestFramework.pas, TextTestRunner.pas,
	versioninfo.inc: Moved new ITestListener methods to ITestListenerX
	to try not to break existing tools. Should merge before a release.
	
	Still can't get WANT to run DUnit test dlls.

2003-03-20 05:03  juanco

	* src/versioninfo.inc: routine

2003-03-19 22:23  juanco

	* doc/API.zip: removed outdated API docs

2003-03-19 13:51  chrismo

	* src/GUITestRunner.pas: Made [Start/End][Test/Suite] virtual
	methods in the GUIRunner

2003-03-18 20:35  jonbsfnet

	* src/GUITesting.pas: Fixed EnterKey code to eliminate dup
	keypresses.  This solution uses PostMessage so keypresses go
	directly to the app being tested.

2003-03-18 20:33  jonbsfnet

	* tests/: UnitTests.dpr, UnitTestGUITesting.pas,
	UnitTestGUITesting.dfm: Created unit tests for TGUITestCase.  Test
	EnterKeyInto and Tab.

2003-03-16 21:04  neuromancer

	* src/GUITesting.pas: shift keys fixed in CLX

2003-03-13 21:04  chrismo

	* src/: GUITestRunner.pas, TestFramework.pas, TextTestRunner.pas:
	Added StartSuite and EndSuite to listener interface and implemented
	these methods where necessary. GUITestListener and TextTestListener
	do not do anything with these, but they are there for subclass
	support.

2003-03-10 01:08  juanco

	* want.xml: added XML tag

2003-03-09 18:38  juanco

	* src/TestFramework.pas, src/versioninfo.inc, src/versioninfo.rc,
	src/versioninfo.res, tests/UnitTestFramework.pas: incorporated bin
	and hex comparisons by Kristofer Skaug. All tests pass.

2003-03-01 02:04  juanco

	* ChangeLog, want.xml, src/versioninfo.inc: changed version number
	to see if email notifications work

2003-03-01 00:32  juanco

	* admin/loginfo: mistyped the mailing list name

2003-03-01 00:25  juanco

	* admin/syncmail: local copy of syncmail no longer needed

2003-03-01 00:24  juanco

	* admin/loginfo: log cvs activity to the developers mailing list

2003-02-15 04:25  juanco

	* doc/: cvs_access.html, images/wincvs_setup_1.jpg,
	images/wincvs_setup_2.jpg: CVS write access howto

2003-01-27 12:48  pvspain

	* Contrib/DUnitWizard/: README.txt, Common/EPCOTAUtils50.cfg,
	Common/EPCOTAUtils50.dof, Common/EPCOTAUtils50.dpk,
	Common/EPCOTAUtils60.cfg, Common/EPCOTAUtils60.dof,
	Common/EPCOTAUtils60.dpk, Common/EPCOTAUtils70.cfg,
	Common/EPCOTAUtils70.dof, Common/EPCOTAUtils70.dpk,
	Common/XP_OTACreators.pas, Common/XP_OTAFileNamer.dfm,
	Common/XP_OTAFileNamer.pas, Common/XP_OTAUtils.pas,
	Common/XP_OTAWizards.pas, Wizard/DUnitWizardProjectGroup50.bpg,
	Wizard/DUnitWizardProjectGroup60.bpg,
	Wizard/DUnitWizardProjectGroup70.bpg, Wizard/EPCDUnitWizard50.cfg,
	Wizard/EPCDUnitWizard50.dof, Wizard/EPCDUnitWizard50.dpk,
	Wizard/EPCDUnitWizard60.cfg, Wizard/EPCDUnitWizard60.dof,
	Wizard/EPCDUnitWizard60.dpk, Wizard/EPCDUnitWizard70.cfg,
	Wizard/EPCDUnitWizard70.dof, Wizard/EPCDUnitWizard70.dpk,
	Wizard/ProjectSource.txt, Wizard/TestCaseSource.txt,
	Wizard/XPDUnitConstants.pas, Wizard/XPDUnitMenuWizard.pas,
	Wizard/XPDUnitProjectWizard.pas, Wizard/XPDUnitProjectWizard.rc,
	Wizard/XPDUnitTestCaseWizard.pas, Wizard/XPDUnitTestCaseWizard.rc,
	Wizard/XPDUnitWizard.pas, Wizard/dunit_project.bmp,
	Wizard/dunit_project16.ico, Wizard/dunit_project256.ico,
	Wizard/dunit_testcase.bmp, Wizard/dunit_testcase16.ico,
	Wizard/dunit_testcase256.ico: DUnitWizard IDE plug-in v1.0 imported

2003-01-08 14:33  juanco

	* want.xml, doc/README.html, doc/README_CHT.html,
	src/DUnitMainForm.pas, src/DunitAbout.pas, src/GUITestRunner.pas,
	src/GUITesting.pas, src/QGUITestRunner.pas, src/TestExtensions.pas,
	src/TestFramework.pas, src/TestModules.pas, src/TextTestRunner.pas,
	src/dunit.dpr, src/versioninfo.inc, tests/DUnitTestLib.dpr,
	tests/UnitTestExtensions.pas, tests/UnitTestFramework.pas,
	tests/UnitTestGUITestRunner.pas, tests/UnitTestGUITesting.pas,
	tests/UnitTests.dpr: updated copyrights

2002-12-06 17:22  uberto

	* src/TestFramework.pas: Removed CheckEqual with Currency params
	Changed Double params in CheckEqual in extended

2002-12-03 16:50  juanco

	* doc/images/: environment_options.jpg, project_options.jpg,
	screenshot.png, screenshot_running.png, tw.jpg, usa.gif,
	xtao128.png, xtao128_new.png, xtao16-16.png, xtao16-256.png,
	xtao16.png, xtao256.png, xtao32-256.png, xtao32.png, xtao64.png:
	checked in images again to try to resolve problems with reported by
	Kris

2002-12-02 18:39  neuromancer

	* src/QGUITestRunner.pas: Kylix 3 compatybility

2002-12-01 12:19  neuromancer

	* src/TestFramework.pas: compatybility with D4 and D5

2002-11-29 22:30  neuromancer

	* readme-kylix.txt: notes

2002-11-06 17:42  juanco

	* Contrib/XPGen/frmXPGenOpts.dfm, Contrib/XPGen/testXpgen.res,
	Contrib/XPGen/testunit.dfm, Contrib/XPGen/xpgen.res,
	src/DUnitMainForm.dfm, src/DunitAbout.dfm, src/GUITestRunner.dfm,
	src/GUITesting.pas, src/versioninfo.inc: updates necessary for a
	build on a different machine

2002-11-06 14:26  juanco

	* Contrib/XPGen/frmXPGenOpts.dfm, Contrib/XPGen/testXpgen.res,
	Contrib/XPGen/testunit.dfm, Contrib/XPGen/xpgen.res,
	Contrib/momus/orig tests/TestsByNumber.zip, doc/API.zip,
	doc/images/environment_options.jpg, doc/images/project_options.jpg,
	doc/images/screenshot.png, doc/images/screenshot_running.png,
	doc/images/tw.jpg, doc/images/usa.gif, doc/images/xtao128.png,
	doc/images/xtao128_new.png, doc/images/xtao16-16.png,
	doc/images/xtao16-256.png, doc/images/xtao16.png,
	doc/images/xtao256.png, doc/images/xtao32-256.png,
	doc/images/xtao32.png, doc/images/xtao64.png,
	examples/structure/diffunit/UnitStructure.png,
	src/DUnitMainForm.dfm, src/DunitAbout.dfm, src/GUITestRunner.dfm,
	src/images/logo.bmp: routine

2002-10-30 16:23  neuromancer

	* examples/structure/: diffunit/Project1Test.dpr,

Contrib/DUnitWizard/Common/XP_OTACreators.pas
unit XP_OTACreators;

{
 $Source: /cvsroot/dunit/dunit/Contrib/DUnitWizard/Common/XP_OTACreators.pas,v $
 $Revision: 1.1 $
 $Date: 2003/01/27 12:48:27 $
 Last amended by $Author: pvspain $
 $State: Exp $

 XP_OTAProjectCreator:
 Base class for IOTACreator and descendant interface implementations

 Copyright (c) 2001 by The Excellent Programming Company Pty Ltd
 (Australia) (ABN 27 005 394 918).

 Contact Paul Spain via email: paul@xpro.com.au

 This unit is free software; you can redistribute it and/or
 modify it under the terms of the GNU Lesser General Public
 License as published by the Free Software Foundation; either
 version 2.1 of the License, or (at your option) any later version.

 This unit is distributed in the hope that it will be useful,
 but WITHOUT ANY WARRANTY; without even the implied warranty of
 MERCHANTABILITY or FITNESS FOR A PARTICULAR PURPOSE.  See the GNU
 Lesser General Public License for more details.

 You should have received a copy of the GNU Lesser General Public
 License along with this unit; if not, the license can be viewed at:
 http://www.gnu.org/copyleft/lesser.html
 or write to the Free Software Foundation, Inc., 59 Temple Place, Suite 330,
 Boston, MA  02111-1307  USA
 }

interface

uses
  ToolsAPI;

//////////////////////////////////////////////////////////////////////////////
///     TXP_OTACreator declaration
//////////////////////////////////////////////////////////////////////////////

type

 TXP_OTACreator = class (TInterfacedObject, IOTACreator)
 protected

   // IOTACreator implementation

   function GetCreatorType: string; virtual; abstract;
   function GetExisting: Boolean; virtual;
   function GetFileSystem: string; virtual;
   function GetOwner: IOTAModule; virtual;
   function GetUnnamed: Boolean; virtual;

 end;

//////////////////////////////////////////////////////////////////////////////
///     TXP_OTAModuleCreator declaration
//////////////////////////////////////////////////////////////////////////////

type

  TXP_OTAModuleCreator = class (TXP_OTACreator, IOTAModuleCreator)
  protected

    FAbsoluteUnitName: string;

    // IOTACreator implementation

    function GetCreatorType: string; override;
    function GetOwner: IOTAModule; override;

    // IOTAModuleCreator implementation

    function GetAncestorName: string; virtual;
    function GetImplFileName: string; virtual;
    function GetIntfFileName: string;
    function GetFormName: string; virtual;
    function GetMainForm: Boolean; virtual;
    function GetShowForm: Boolean; virtual;
    function GetShowSource: Boolean; virtual;
    function NewIntfSource(
      const ModuleIdent, FormIdent, AncestorIdent: string): IOTAFile;
    procedure FormCreated(const FormEditor: IOTAFormEditor); virtual;

    // Subclasses must override NewFormFile() and
    // NewImplSource()

    // Create and return the Form resource for this new module if applicable
    function NewFormFile(const FormIdent,
      AncestorIdent: string): IOTAFile; virtual; abstract;
    // Create and return the Implementation source for this module.
    // (Delphi unit)
    function NewImplSource(const ModuleIdent, FormIdent,
      AncestorIdent: string): IOTAFile; virtual; abstract;


  public

    constructor Create(const AUnitName: string = '');
  end;

//////////////////////////////////////////////////////////////////////////////
///     TXP_OTAUnitCreator declaration
//////////////////////////////////////////////////////////////////////////////

type

  TXP_OTAUnitCreator = class (TXP_OTAModuleCreator)
  protected

    // IOTACreator implementation

    function GetCreatorType: string; override;

    // IOTAModuleCreator implementation

    function NewFormFile(const FormIdent,
      AncestorIdent: string): IOTAFile; override;

    // Subclasses must override NewImplSource()

   end;
      

//////////////////////////////////////////////////////////////////////////////
///     TXP_OTAProjectCreator declaration
//////////////////////////////////////////////////////////////////////////////

type

  TXP_OTAProjectCreator = class (TXP_OTACreator, IOTAProjectCreator)
  protected

    FAbsoluteProjectName: string;

    // IOTACreator implementation

    function GetCreatorType: string; override;
    function GetOwner: IOTAModule; override;
    function GetUnnamed: Boolean; override;

    // IOTAProjectCreator implementation

    function GetFileName: string; virtual;
    function GetOptionFileName: string; virtual;
    function GetShowSource: Boolean; virtual;
    procedure NewDefaultModule; virtual;
    function NewOptionSource(const ProjectName: string): IOTAFile; virtual;
    procedure NewProjectResource(const Project: IOTAProject); virtual;

    // Subclasses must override NewProjectSource()

    function NewProjectSource(
     const ProjectName: string): IOTAFile; virtual; abstract;

  public

    constructor Create(const AProjectName: string = '');
  end;


implementation

uses
  SysUtils,           // ExpandFileName() ...
  XP_OTAUtils;        // GetCurrentProjectGroup()

const CVSID: string = '$Header: /cvsroot/dunit/dunit/Contrib/DUnitWizard/Common/XP_OTACreators.pas,v 1.1 2003/01/27 12:48:27 pvspain Exp $';

//////////////////////////////////////////////////////////////////////////////
///     TXP_OTACreator implementation
//////////////////////////////////////////////////////////////////////////////

function TXP_OTACreator.GetExisting: Boolean;
begin
  // New module
  Result := false;
end;


function TXP_OTACreator.GetFileSystem: string;
begin
  // Use default (non-virtual) file system
  Result := '';
end;


function TXP_OTACreator.GetOwner: IOTAModule;
begin
  // No owner
  Result := nil;
end;

function TXP_OTACreator.GetUnnamed: Boolean;
begin
  // Prompt with "Save As..."  when the user closes this module.
  Result := true;
end;

//////////////////////////////////////////////////////////////////////////////
///     TXP_OTAModuleCreator implementation
//////////////////////////////////////////////////////////////////////////////

constructor TXP_OTAModuleCreator.Create(const AUnitName: string);
begin
  inherited Create;
  FAbsoluteUnitName := SysUtils.Trim(AUnitName);

  if FAbsoluteUnitName <> '' then
  begin
    // Ensure we have an absolute file specification for GetFileName()
    FAbsoluteUnitName := SysUtils.ExpandFileName(FAbsoluteUnitName);
    // Create as a PAS file
    SysUtils.ChangeFileExt(FAbsoluteUnitName, '.pas');
  end;

end;  

procedure TXP_OTAModuleCreator.FormCreated(
  const FormEditor: IOTAFormEditor);
begin
  // Called when the new form/datamodule/custom module is created
end;

function TXP_OTAModuleCreator.GetAncestorName: string;
begin
  // Return the Ancestor form name
  Result := 'TForm';
end;

function TXP_OTAModuleCreator.GetCreatorType: string;
begin
  // We supply all required information
  Result := '';
end;

function TXP_OTAModuleCreator.GetFormName: string;
begin
  // Return the form name (without path or extension) with a suffix of 'Form'
  Result := SysUtils.ChangeFileExt(
    SysUtils.ExtractFileName(FAbsoluteUnitName), '') + 'Form';
end;

function TXP_OTAModuleCreator.GetImplFileName: string;
begin
  // Return the implementation filename, or blank to have the IDE create a new
  // unique one. (C++ .cpp file or Delphi unit)
  // Must be an absolute path
  Result  := FAbsoluteUnitName;
end;

function TXP_OTAModuleCreator.GetIntfFileName: string;
begin
 // Return the interface filename, or blank to have the IDE create a new
 // unique one.  (C++ header) - not applicable to Delphi
 Result := '';
end;

function TXP_OTAModuleCreator.GetMainForm: Boolean;
begin
  // Return True to Make this module the main form of the given Owner/Project
  Result := false;
end;

function TXP_OTAModuleCreator.GetOwner: IOTAModule;
var
  Project: IOTAProject;

begin
  // Make the current project the owner of this module
  XP_OTAUtils.GetCurrentProject(Project);
  Result := Project;  
end;

function TXP_OTAModuleCreator.GetShowForm: Boolean;
begin
  // Return True to show the form 
  Result := true;
end;

function TXP_OTAModuleCreator.GetShowSource: Boolean;
begin
  // Return True to show the source
  Result := true;
end;

function TXP_OTAModuleCreator.NewIntfSource(const ModuleIdent, FormIdent,
  AncestorIdent: string): IOTAFile;
begin
  // Create and return the Interface (C++ header) source for this module
  Result := nil;
end;

//////////////////////////////////////////////////////////////////////////////
///     TXP_OTAUnitCreator implementation
//////////////////////////////////////////////////////////////////////////////

function TXP_OTAUnitCreator.GetCreatorType: string;
begin
  Result := ToolsAPI.sUnit;
end;

function TXP_OTAUnitCreator.NewFormFile(const FormIdent,
  AncestorIdent: string): IOTAFile;
begin
  // No form file required for a unit...
  Result := nil;
end;

//////////////////////////////////////////////////////////////////////////////
///     TXP_OTAProjectCreator implementation
//////////////////////////////////////////////////////////////////////////////

constructor TXP_OTAProjectCreator.Create(const AProjectName: string);
begin
  inherited Create;
  FAbsoluteProjectName := SysUtils.Trim(AProjectName);

  if FAbsoluteProjectName <> '' then
  begin
    // Ensure we have an absolute file specification for GetFileName()
    FAbsoluteProjectName := SysUtils.ExpandFileName(FAbsoluteProjectName);

    // Create as an application (or library) if extension not provided
    if SysUtils.ExtractFileExt(FAbsoluteProjectName) = '' then
      FAbsoluteProjectName := FAbsoluteProjectName + '.dpr';

  end;

end;

function TXP_OTAProjectCreator.GetCreatorType: string;
begin
  // Invoke default application behaviour for the hard methods!
  Result := ToolsAPI.sApplication;
end;

function TXP_OTAProjectCreator.GetFileName: string;
begin
  Result := FAbsoluteProjectName;
end;

function TXP_OTAProjectCreator.GetOptionFileName: string;
begin
  // Invoke default application behaviour since we set ToolsAPI.sApplication
  // in GetCreatorType()
  Result := '';
end;

function TXP_OTAProjectCreator.GetOwner: IOTAModule;
var
  ProjectGroup: IOTAProjectGroup;

begin
  // Add to current project group if available - ProjectGroup is nil
  // on failure
  XP_OTAUtils.GetCurrentProjectGroup(ProjectGroup);
  Result := ProjectGroup;
end;

function TXP_OTAProjectCreator.GetShowSource: Boolean;
begin
  // Show the project source file in the IDE source editor
  Result := true;
end;

function TXP_OTAProjectCreator.GetUnnamed: Boolean;
begin
  // Prompt with "Save As" if FProjectName is undefined
  Result := FAbsoluteProjectName = '';
end;

procedure TXP_OTAProjectCreator.NewDefaultModule;
begin
  // Invoke default application behaviour since we set ToolsAPI.sApplication
  // in GetCreatorType()
end;

function TXP_OTAProjectCreator.NewOptionSource(
  const ProjectName: string): IOTAFile;
begin
  // Invoke default application behaviour since we set ToolsAPI.sApplication
  // in GetCreatorType()
  Result := nil;
end;

procedure TXP_OTAProjectCreator.NewProjectResource(const Project: IOTAProject);
begin
  // Invoke default application behaviour since we set ToolsAPI.sApplication
  // in GetCreatorType()
end;

end.


